### ISSUE TITLE
[BUG] fresh bare-domain installation, not able to auth - dovecot

### ISSUE THREAD

**Issue by smargold476** [docker-mailserver/docker-mailserver#2678]
### Miscellaneous first checks

- [X] I checked that all ports are open and not blocked by my ISP / hosting provider.
- [X] I know that SSL errors are likely the result of a wrong setup on the user side and not caused by DMS itself. I'm confident my setup is correct.

### Affected Component(s)

dovecot

### What happened and when does this occur?

```Markdown
fresh setup (bare/naked-domain-setup)
configured external certs for my base-domain -> mydomain.org (looks like that is not a issue here)
create a user
try to login with thunderbird
password-check timeout (login not successful)

see the logs
```

### What did you expect to happen?

```Markdown
login successful
```

### How do we replicate the issue?

```Markdown
for bare-domain, i configured 2 things beside default:
 - in `docker-compose.yml` # before `domainname:` and set `hostname: mydomain.org`
 - create `config/postfix-main.cf` and add `mydestination = localhost.$mydomain, localhost`

1. fresh setup (bare/naked-domain-setup)
2. create a user (username `test` password `password`
3. try to login with thunderbird
```

### DMS version

11.0.0

### What operating system is DMS running on?

Linux

### What instruction set architecture is DMS running on?

x86_64 / AMD64

### What container orchestration tool are you using?

Docker Compose

### docker-compose.yml

```yml
services:
  mailserver:
    image: docker.io/mailserver/docker-mailserver:latest
    container_name: mailserver
    hostname: mydomain.org
    env_file: mailserver.env # there is SSL_TYPE, SSL_CERT_PATH and SSL_KEY_PATH defined, beside defaults
    ports:
      - "25:25"    # SMTP  (explicit TLS => STARTTLS)
      - "143:143"  # IMAP4 (explicit TLS => STARTTLS)
      - "465:465"  # ESMTP (implicit TLS)
      - "587:587"  # ESMTP (explicit TLS => STARTTLS)
      - "993:993"  # IMAP4 (implicit TLS)
    volumes:
      - /mountpointInLocalSystem/mail-data/:/var/mail/
      - /mountpointInLocalSystem/mail-state/:/var/mail-state/
      - /mountpointInLocalSystem/mail-logs/:/var/log/mail/
      - /mountpointInLocalSystem/config/:/tmp/docker-mailserver/
      - /etc/localtime:/etc/localtime:ro
      - /mountpointInLocalSystem/ssl/:/tmp/dms/custom-certs/:ro
    restart: always
    stop_grace_period: 1m
    cap_add:
      - NET_ADMIN
    healthcheck:
      test: "ss --listening --tcp | grep -P 'LISTEN.+:smtp' || exit 1"
      timeout: 3s
      retries: 0
```

### Relevant log output

```bash
[   INF   ]  Welcome to docker-mailserver 11.0.0
[   INF   ]  Initializing setup
[   INF   ]  Checking configuration
[   INF   ]  Configuring mail server
grep: /tmp/docker-mailserver/postfix-accounts.cf: No such file or directory
[ WARNING ]  You need at least one email account to start Dovecot (120s left for account creation before shutdown)
grep: /tmp/docker-mailserver/postfix-accounts.cf: No such file or directory
[ WARNING ]  You need at least one email account to start Dovecot (110s left for account creation before shutdown)
grep: /tmp/docker-mailserver/postfix-accounts.cf: No such file or directory
[ WARNING ]  You need at least one email account to start Dovecot (100s left for account creation before shutdown)
[ WARNING ]  No DKIM key(s) provided - check the documentation on how to get your keys
[   INF   ]  Post-configuration checks
[   INF   ]  Starting miscellaneous tasks
[   INF   ]  Starting daemons
[   INF   ]  mydomain.org is up and running
Jul  7 22:41:28 mydomain postfix/postdrop[535]: warning: unable to look up public/pickup: No such file or directory
Jul  7 22:41:28 mydomain amavis[527]: starting. /usr/sbin/amavisd-new at mydomain.org amavisd-new-2.11.1 (20181009), Unicode aware, LC_CTYPE="C.UTF-8"
Jul  7 22:41:28 mydomain amavis[527]: perl=5.032001, user=, EUID: 112 (112);  group=, EGID: 114 114 (114 114)
Jul  7 22:41:28 mydomain amavis[527]: Net::Server: Group Not Defined.  Defaulting to EGID '114 114'
Jul  7 22:41:28 mydomain amavis[527]: Net::Server: User Not Defined.  Defaulting to EUID '112'
Jul  7 22:41:28 mydomain amavis[527]: No ext program for   .zoo, tried: zoo
Jul  7 22:41:28 mydomain amavis[527]: No ext program for   .doc, tried: ripole
Jul  7 22:41:28 mydomain amavis[527]: No decoder for       .F   
Jul  7 22:41:28 mydomain amavis[527]: No decoder for       .doc 
Jul  7 22:41:28 mydomain amavis[527]: No decoder for       .zoo 
Jul  7 22:41:31 mydomain postfix/master[1707]: daemon started -- version 3.5.6, configuration /etc/postfix
Jul  7 22:41:31 mydomain postfix/pickup[1712]: <mydockerlocalhostname>: uid=0 from=<root>
Jul  7 22:41:31 mydomain postfix/cleanup[1718]: warning: connect to Milter service inet:localhost:8891: Cannot assign requested address
Jul  7 22:41:31 mydomain postfix/cleanup[1718]: <mydockerlocalhostname>: message-id=<20220707204131.<mydockerlocalhostname>@mydomain.org>
Jul  7 22:41:31 mydomain postfix/qmgr[1713]: <mydockerlocalhostname>: from=<root@mydomain.org>, size=711, nrcpt=1 (queue active)
Jul  7 22:41:31 mydomain postfix/local[1732]: warning: required alias not found: postmaster
Jul  7 22:41:31 mydomain postfix/local[1732]: <mydockerlocalhostname>: to=<postmaster@mydomain.org>, relay=local, delay=2.9, delays=2.9/0.01/0/0, dsn=2.0.0, status=sent (discarded)
Jul  7 22:41:31 mydomain postfix/qmgr[1713]: <mydockerlocalhostname>: removed
Jul  7 22:42:06 mydomain postfix/submission/smtpd[1856]: connect from <mythunderbirdsystem>[83.83.83.83]
Jul  7 22:42:06 mydomain postfix/submission/smtpd[1856]: warning: connect to Milter service inet:localhost:8891: Cannot assign requested address
Jul  7 22:42:06 mydomain postfix/submission/smtpd[1863]: connect from <mythunderbirdsystem>[83.83.83.83]
Jul  7 22:42:06 mydomain postfix/submission/smtpd[1863]: warning: connect to Milter service inet:localhost:8891: Cannot assign requested address
Jul  7 22:42:06 mydomain postfix/submission/smtpd[1866]: connect from <mythunderbirdsystem>[83.83.83.83]
Jul  7 22:42:06 mydomain postfix/submission/smtpd[1866]: warning: connect to Milter service inet:localhost:8891: Cannot assign requested address
Jul  7 22:42:06 mydomain postfix/submission/smtpd[1867]: connect from <mythunderbirdsystem>[83.83.83.83]
Jul  7 22:42:06 mydomain postfix/submission/smtpd[1867]: warning: connect to Milter service inet:localhost:8891: Cannot assign requested address
Jul  7 22:42:06 mydomain dovecot: imap-login: Aborted login (no auth attempts in 0 secs): user=<>, rip=83.83.83.83, lip=172.26.0.2, session=<mlVrHj3j8/5d3oET>
Jul  7 22:42:06 mydomain dovecot: imap-login: Aborted login (no auth attempts in 0 secs): user=<>, rip=83.83.83.83, lip=172.26.0.2, session=<mlVrHj3j7v5d3oET>
Jul  7 22:42:06 mydomain dovecot: imap-login: Aborted login (no auth attempts in 0 secs): user=<>, rip=83.83.83.83, lip=172.26.0.2, session=<Y1ZrHj3j8f5d3oET>
Jul  7 22:42:06 mydomain postfix/submission/smtpd[1856]: improper command pipelining after EHLO from <mythunderbirdsystem>[83.83.83.83]: QUIT\r\n
Jul  7 22:42:06 mydomain dovecot: imap-login: Aborted login (no auth attempts in 0 secs): user=<>, rip=83.83.83.83, lip=172.26.0.2, session=<R1hrHj3j7P5d3oET>
Jul  7 22:42:06 mydomain postfix/submission/smtpd[1856]: disconnect from <mythunderbirdsystem>[83.83.83.83] ehlo=1 quit=1 commands=2
Jul  7 22:42:06 mydomain postfix/submission/smtpd[1863]: disconnect from <mythunderbirdsystem>[83.83.83.83] ehlo=1 quit=1 commands=2
Jul  7 22:42:06 mydomain postfix/submission/smtpd[1866]: improper command pipelining after EHLO from <mythunderbirdsystem>[83.83.83.83]: QUIT\r\n
Jul  7 22:42:06 mydomain postfix/submission/smtpd[1866]: disconnect from <mythunderbirdsystem>[83.83.83.83] ehlo=1 quit=1 commands=2
Jul  7 22:42:06 mydomain postfix/submission/smtpd[1867]: improper command pipelining after EHLO from <mythunderbirdsystem>[83.83.83.83]: QUIT\r\n
Jul  7 22:42:06 mydomain postfix/submission/smtpd[1867]: disconnect from <mythunderbirdsystem>[83.83.83.83] ehlo=1 quit=1 commands=2
Jul  7 22:42:20 mydomain postfix/submission/smtpd[1856]: connect from <mythunderbirdsystem>[83.83.83.83]
Jul  7 22:42:20 mydomain postfix/submission/smtpd[1856]: warning: connect to Milter service inet:localhost:8891: Cannot assign requested address
Jul  7 22:42:20 mydomain postfix/submission/smtpd[1856]: disconnect from <mythunderbirdsystem>[83.83.83.83] ehlo=1 quit=1 commands=2
Jul  7 22:42:20 mydomain dovecot: imap-login: Aborted login (no auth attempts in 0 secs): user=<>, rip=83.83.83.83, lip=172.26.0.2, session=<Y7hAHz3jFf9d3oET>
Jul  7 22:42:23 mydomain dovecot: auth: passwd-file(testuser,83.83.83.83,<1jtpHz3jF/9d3oET>): unknown user (SHA1 of given password: <testsha>)
Jul  7 22:42:29 mydomain dovecot: auth: passwd-file(testuser,83.83.83.83,<1jtpHz3jF/9d3oET>): unknown user (SHA1 of given password: <testsha>)
Jul  7 22:42:35 mydomain dovecot: auth: passwd-file(testuser,83.83.83.83,<1jtpHz3jF/9d3oET>): unknown user (SHA1 of given password: <testsha>)
Jul  7 22:42:37 mydomain dovecot: imap-login: Disconnected (auth failed, 3 attempts in 14 secs): user=<testuser>, method=PLAIN, rip=83.83.83.83, lip=172.26.0.2, TLS, session=<1jtpHz3jF/9d3oET>
Jul  7 22:43:11 mydomain postfix/master[1707]: terminating on signal 15
Jul  7 22:43:11 mydomain dovecot: master: Warning: Killed with signal 15 (by pid=7 uid=0 code=kill)
Jul  7 22:43:12 mydomain dovecot: master: Dovecot v2.3.13 (89f716dc2) starting up for imap, lmtp
Jul  7 22:43:13 mydomain postfix/master[3813]: daemon started -- version 3.5.6, configuration 
```
```


### Other relevant information

_No response_

### What level of experience do you have with Docker and mail servers?

- [X] I am inexperienced with docker
- [X] I am inexperienced with mail servers
- [ ] I am uncomfortable with the CLI

### Code of conduct

- [X] I have read this project's [Code of Conduct](https://github.com/docker-mailserver/docker-mailserver/blob/master/CODE_OF_CONDUCT.md) and I agree
- [X] I have read the [README](https://github.com/docker-mailserver/docker-mailserver/blob/master/README.md) and the [documentation](https://docker-mailserver.github.io/docker-mailserver/edge/) and I searched the [issue tracker](https://github.com/docker-mailserver/docker-mailserver/issues?q=is%3Aissue) but could not find a solution

### Improvements to this form?

_No response_

**Comment by smargold476** [comment]
Additial Info:
`doveadm auth test -a /var/spool/postfix/private/auth testuser testpassword`
executed inside the container returns:

```
passdb: testuser auth failed
extra fields:
  user=testuser
```

**Comment by polarathene** [comment]
Hello, before putting time into looking into this issue, I'd like to state that the usage of bare-domain config causing problems is not uncommon.

Often though, the desire to configure for a bare-domain is a misunderstanding by the user, and I would like to start addressing the actual problem, rather than fixing support for the wrong thing.

---

Do you actually want your mail-server hosted and reachable at `example.com`, or is your actual intent for mail to be exchanged with `@example.com` not `@mail.example.com`?

Often there is a misunderstanding that `hostname: mail.example.com` equates to `user@mail.example.com`. You should find that by default you'd actually be exchanging via `user@example.com`, `mail.example.com` is for your DNS records so that `@example.com` can know where the mail server is located.

Your mail server does not have to belong to `example.com`. You could have `mail.example.net` handle mail, and allow receiving and sending mail on behalf of many other domains you own.

---

If your interest in a bare-domain remains, could you please share the use-case? If so I will attempt to reproduce your issue.

**Comment by smargold476** [comment]
Hi @polarathene,

thanks for your clarification, I understand that many user do not get that correctly, maybe it would be nice to see your text in the FAQ-part regarding bare-domain-config?

I guess my interest is still there, i own a server with the hostname `example.com` so I guess I need to follow the bare-domain-config instructions? I also created a MX-record pointing to example.com.

Would it be possible to create a cname-record `mail.example.com` and point it to my `example.com`-server plus create the MX-Record pointing to mail.example.com (then i would not need the bare-domain-config...)?
If that would be a "supported" workaround, I suggest to add that information in the FAQ, i thougt about this solution, but then I saw some commands like `hostname` inside the project....

Thanks for your quick answer!

**Comment by polarathene** [comment]
> I guess my interest is still there, i own a server with the hostname `example.com` so I guess I need to follow the bare-domain-config instructions? I also created a MX-record pointing to `example.com`.

That sounds like the confusion I'm talking about? :sweat_smile: 

Can you explain what advantage you're expecting from a bare-domain config for your mail server?

---

> Would it be possible to create a cname-record `mail.example.com` and point it to my `example.com`-server plus create the MX-Record pointing to `mail.example.com` (then i would not need the bare-domain-config...)?

No, IIRC it is ill-advised to use a CNAME record instead of an A/AAAA record for your MTA when it comes to resolving the IP. What advantage are you expecting from the CNAME record here? Is there a problem you're trying to solve that requires a bare-domain setup?

Your `example.com` should have an MX record that points to the A record of the MTA (`docker-mailserver`). MX record is for looking up where to deliver mail to (_receiving/inbound mail to your MX_). That is `user@example.com` will query DNS for MX record of `example.com` and that may return the A record `mail.example.com` (_or at a different domain such as `mail.example.net` if you like_), which will then query the IP.

When you send mail, `docker-mailserver` / Postfix would announce itself as `mail.example.com` to another mail service it's sending to, where the other side will possibly check that you are authorized to send mail on behalf of the mail sender (_the SPF record_). The MX record does not matter for this exchange.

Likewise if you allow another service such as SendGrid (_or another `docker-mailserver` instance_) to receive mail and send it to the destination on your behalf, it will announce itself with their own MTA hostname, unrelated to your domain, but SPF record still checked for if the mail is legitimately authorized to be sent on your behalf, or likely a spammer if not.

---

> I understand that many user do not get that correctly, maybe it would be nice to see your text in the FAQ-part regarding bare-domain-config?

Did any of that clarify the configuration better?

It's been a while for me, but I remember it being confusing myself when I started. If it has made a difference it would be good to know what was most helpful, and any misunderstanding you had as I believe it's common for our users, and I would like to document it better.

Or have I misunderstood, and you still want a bare-domain setup for some other reason?

---

> If that would be a "supported" workaround, I suggest to add that information in the FAQ, i thougt about this solution, but then I saw some commands like `hostname` inside the project....

Our bare-domain support (and docs related to it) I believe was contributed with a similar misunderstanding at the time. We've had similar legacy code/support that I'd also like to remove in future, such as `domainname:` (_which is unrelated to `example.com` in your DNS/registrar. It refers to NIS domains, only `hostname:` should be used IMO_).

**Comment by georglauterbach** [comment]
> > I guess my interest is still there, i own a server with the hostname `example.com` so I guess I need to follow the bare-domain-config instructions? I also created a MX-record pointing to `example.com`.
> 
> That sounds like the confusion I'm talking about? sweat_smile

Haha, exactly my thought when reading this @polarathene :D

**Comment by smargold476** [comment]
> > I guess my interest is still there, i own a server with the hostname `example.com` so I guess I need to follow the bare-domain-config instructions? I also created a MX-record pointing to `example.com`.
> 
> That sounds like the confusion I'm talking about? 😅
> 
> Can you explain what advantage you're expecting from a bare-domain config for your mail server?

I'm not looking for a advantage, but for the correct config to get it up&running.
If it is possible to have a server with the Hostname `example.com`, add a A-Record `mail.example.com` pointing to `example.com`'s IP and then use the default configuration without the need of a bare-domain-conf, its fine for me.


> > Would it be possible to create a cname-record `mail.example.com` and point it to my `example.com`-server plus create the MX-Record pointing to `mail.example.com` (then i would not need the bare-domain-config...)?
> 
> No, IIRC it is ill-advised to use a CNAME record instead of an A/AAAA record for your MTA when it comes to resolving the IP. What advantage are you expecting from the CNAME record here? Is there a problem you're trying to solve that requires a bare-domain setup?
> 
> Your `example.com` should have an MX record that points to the A record of the MTA (`docker-mailserver`). MX record is for looking up where to deliver mail to (_receiving/inbound mail to your MX_). That is `user@example.com` will query DNS for MX record of `example.com` and that may return the A record `mail.example.com` (_or at a different domain such as `mail.example.net` if you like_), which will then query the IP.
> 
> When you send mail, `docker-mailserver` / Postfix would announce itself as `mail.example.com` to another mail service it's sending to, where the other side will possibly check that you are authorized to send mail on behalf of the mail sender (_the SPF record_). The MX record does not matter for this exchange.
> 
> Likewise if you allow another service such as SendGrid (_or another `docker-mailserver` instance_) to receive mail and send it to the destination on your behalf, it will announce itself with their own MTA hostname, unrelated to your domain, but SPF record still checked for if the mail is legitimately authorized to be sent on your behalf, or likely a spammer if not.
> 

I just work with cnames to avoid chaning a lot of subdomains when exchaning the public-ip, so creating a A-record is no problem at all.


> > I understand that many user do not get that correctly, maybe it would be nice to see your text in the FAQ-part regarding bare-domain-config?
> 
> Did any of that clarify the configuration better?
> 
> It's been a while for me, but I remember it being confusing myself when I started. If it has made a difference it would be good to know what was most helpful, and any misunderstanding you had as I believe it's common for our users, and I would like to document it better.
> 
> Or have I misunderstood, and you still want a bare-domain setup for some other reason?
> 
> > If that would be a "supported" workaround, I suggest to add that information in the FAQ, i thougt about this solution, but then I saw some commands like `hostname` inside the project....
> 
> Our bare-domain support (and docs related to it) I believe was contributed with a similar misunderstanding at the time. We've had similar legacy code/support that I'd also like to remove in future, such as `domainname:` (_which is unrelated to `example.com` in your DNS/registrar. It refers to NIS domains, only `hostname:` should be used IMO_).

Updating the docs would be nice if bare-domain-conf is needed at all.
But for my understanding now I cannot see any valid need for a bare-domain-config....
For me it was clear that `@example.com` do not need a server "called" `example.com`, so the last reason i saw for the "bare-domain" config was the real hostname of the server needs be != `example.com`. But if that could be solved by a A-record, what reason could exist where a bare-domain-config is needed?


Thanks for your clarification :)

**Comment by polarathene** [comment]
> If it is possible to have a server with the Hostname `example.com`, add a A-Record `mail.example.com` pointing to `example.com`'s IP and then use the default configuration without the need of a bare-domain-conf, its fine for me.

Use `hostname: mail.example.com`, and have an A record for `mail.example.com` point to the same IP as the A record for `example.com`. Don't use a CNAME record, even though that seems logical, apparently it's not a good idea for mail server compatibility. You can then send mail as `user@example.com` and others can send mail to you to the same address, and your mail server should receive it provided you have the correct MX record.

I have a [lengthy example]((https://github.com/docker-mailserver/docker-mailserver/issues/2601#issuecomment-1135520225) that's rather rough which you could use as reference for what I was talking about earlier. [In another issue](https://github.com/docker-mailserver/docker-mailserver/issues/2649#issuecomment-1159958506) I referenced the example and tried to briefly summarize it. It should clear up any confusion with the DNS records, you can even run the example locally with your own mail client to test it (_you'd need to temporarily configure your network to use the container DNS instead, otherwise run the mail client within a container_).

---

> I just work with cnames to avoid chaning a lot of subdomains when exchaning the public-ip, so creating a A-record is no problem at all.

Yes I understand that. I would normally advise a CNAME record, but with mail servers I recall advice that it can affect mail delivery negatively. Possibly when reverse DNS lookups are performed that expect A records?

---

> Updating the docs would be nice if bare-domain-conf is needed at all.
> But for my understanding now I cannot see any valid need for a bare-domain-config....

Yes I agree. My volunteer time is focused elsewhere, and when I can spare time for this project it's either support or tackling broader tasks that require a fair amount of effort / knowledge to contribute. These are more important for me to resolve as I don't know when/if another contributor will get around to them.

Documentation is important, and I would like to revise it, but as that is easier to contribute (_and this topic is general mail-server configuration_), I am more confident that we can receive contributions from others, so it is a lower priority for me. When I've resolved the bigger concerns I have with the project, I can look into revising the docs. I know they are lacking and not particularly great for users new to working with mail servers.

I don't know of valid use-cases requiring it, and I've had the suspicion that it's likely always a misunderstanding. Unfortunately removing features is a slower process than adding them. A larger change is related to the hostname configuration, but requires improving our tests first, followed by deprecation notice. Similar refactoring is required for the TLS support to be consolidated.

---

> For me it was clear that `@example.com` do not need a server "called" `example.com`, so the last reason i saw for the "bare-domain" config was the real hostname of the server needs be != `example.com`. But if that could be solved by a A-record, what reason could exist where a bare-domain-config is needed?

I assume users get confused (_and that the contribution of the support and to docs as a result of that slipped through review_) with possible beliefs being:

- `mail.example.com` hostname might mean `user@mail.example.com`, which looks unprofessional/ugly to what the user wants, so they think they should configure `hostname: example.com` instead. This is a documentation issue for us to communicate better, but it doesn't help that we're using a Docker config option `hostname` instead of an ENV, which AFAIK shouldn't be needed. The addition of `domainname:` likely only adds to the misunderstanding.
- The other possible confusion comes with TLS and if it should be to `mail.example.com` or `example.com`. To avoid thinking about that, they instead choose `hostname: example.com`, thinking the bare-domain solution should work just as well (_despite our documentation and github issues showing the problems encountered with that_). Again something we could better document.

---

> Thanks for your clarification :)

You're very welcome! :grinning:

**Comment by smargold476** [comment]
In short:
I still have a server configured with the local hostname (`/etc/hosts`) == `example.com`
I added a A-Record for that server IP with hostname `mail.example.com`
I cleaned all the mounted directories of the docker-compose config
I set `hostname: mail.example.com` and `domainname: example.com`

Shouldn't that configuration work? Or do I need to change the "real hostname" (`/etc/hosts`) (than that would be a valid need for the bare-domain-config in case somebody cannot change the hostname or the server is used for other services where this hostname is known...)

Still the same issue....
I dont get it tbh ...

**Comment by polarathene** [comment]
`/etc/hosts` just maps lookup locally to an IP, it's not relevant here? Even on your Docker host OS, it is not required for external servers to connect to yours, they get the IP from the DNS.

Docker manages the containers internal `/etc/hosts` and the hostname/domainnane options usually affect the generation of that file (unless using network mode like `Host`) . 

---

Use only `hostname:` in your config, do not use `domainname:` (outdated documentation AFAIK).

**Comment by smargold476** [comment]
> `/etc/hosts` just maps lookup locally to an IP, it's not relevant here? Even on your Docker host OS, it is not required for external servers to connect to yours, they get the IP from the DNS.
> 
> Docker manages the containers internal `/etc/hosts` and the hostname/domainnane options usually affect the generation of that file (unless using network mode like `Host`) .
> 

Thats what I thougt and why I did not change the hostname (inbeetween I changed the hostname to `mail.example.com` without luck)


> Use only `hostname:` in your config, do not use `domainname:` (outdated documentation AFAIK).
>

I deleted the `domainname:` part, still same issue...

I'll start clean later, maybe there are some caches?

**Comment by polarathene** [comment]
> can somebody verify/reproduce that?

I can look into reproducing some time today possibly. I'll be using a setup [like I detailed here](https://github.com/docker-mailserver/docker-mailserver/issues/2601#issuecomment-1135520225).

You're using `SSL_TYPE=manual` I take it with the volume: `/mountpointInLocalSystem/ssl/:/tmp/dms/custom-certs/:ro`?

What is the host OS? Is your host `/etc/hosts` necessary to configure with the same `example.com` hostname due to something else involved? If so what sort of IP are you directing it to? The public internet one, the container one, or something like `127.0.0.1`?

How have you setup your Thunderbird client to connect to the container? Is Thunderbird being run from the same machine running `docker-mailserver`? As you're using custom certificates instead of LetsEncrypt, are these in your client systems trust store? (or imported into Thunderbird)

I'm thinking your issue may be more related to your client setup. I didn't document that as well in the linked comment, and my recall is a bit foggy, so it would be good to go over that again.

**Comment by smargold476** [comment]
> > can somebody verify/reproduce that?
> 
> I can look into reproducing some time today possibly. I'll be using a setup [like I detailed here](https://github.com/docker-mailserver/docker-mailserver/issues/2601#issuecomment-1135520225).
> 
> You're using `SSL_TYPE=manual` I take it with the volume: `/mountpointInLocalSystem/ssl/:/tmp/dms/custom-certs/:ro`?

Yes, there are my certificates, I configured 

```
SSL_TYPE=manual
SSL_CERT_PATH=/tmp/dms/custom-certs/cert1.pem
SSL_KEY_PATH=/tmp/dms/custom-certs/privkey1.pem
```

and in my path  `/mountpointInLocalSystem/ssl/` are the certificates for `mail.example.com` present. In the logs I cannot see any certificate-issue.

> What is the host OS? Is your host `/etc/hosts` necessary to configure with the same `example.com` hostname due to something else involved? If so what sort of IP are you directing it to? The public internet one, the container one, or something like `127.0.0.1`?
 
There is no reason for local resolving, just to make it clear, my `/etc/hosts` looks like this:

```
127.0.0.1 mail.example.com mail
83.83.83.83 mail.example.com mail
```


> How have you setup your Thunderbird client to connect to the container? Is Thunderbird being run from the same machine running `docker-mailserver`? As you're using custom certificates instead of LetsEncrypt, are these in your client systems trust store? (or imported into Thunderbird)
> 
> I'm thinking your issue may be more related to your client setup. I didn't document that as well in the linked comment, and my recall is a bit foggy, so it would be good to go over that again.

Maybe thats the whole issue, I just did a clean install and connected with my test@example.com with default settings.

![image](https://user-images.githubusercontent.com/105579587/178155657-2b2482e0-4591-4eae-a569-e042b373ea38.png)

Do I need to change something here?

**Comment by smargold476** [comment]
Okay, I double-checked the dovecot-msg and saw the following when i test with user `docker`:

```
docker@mail:/etc/dovecot/conf.d$ doveadm auth test -a /var/spool/postfix/private/auth testuser testpass
doveconf: Fatal: Error in configuration file /etc/dovecot/conf.d/10-ssl.conf line 13: ssl_key: Can't open file /etc/dms/tls/key: Permission denied
```

Maybe the certificates are the problem after all?
Or is that okay because of `/etc/dovecot/conf.d/10-ssl.conf`:

```
# PEM encoded X.509 SSL/TLS certificate and private key. They're opened before
# dropping root privileges, so keep the key file unreadable by anyone but root.
# These [snakeoil files actually exist](https://askubuntu.com/questions/396120/what-is-the-purpose-of-the-ssl-cert-snakeoil-key), but shouldn't ever be used in production!
# As `SSL_TYPE` env is required by docker-mailserver, these "snakeoil" files will be replaced on container startup.
ssl_cert = </etc/dms/tls/cert
ssl_key = </etc/dms/tls/key
```

what results in `/etc/dms/tls/`:

```
-rw-r--r-- 1 root root 1846 Jul 10 19:40 cert
-rw------- 1 root root 1704 Jul 10 19:40 key
```



Anyway, permissions in my fs (`/mountpointInLocalSystem/ssl/`) looks like this:

```
-rw-r--r-- 1 mylocaluser mylocalgroup1846 Jul  7 07:49 cert1.pem
-rw------- 1 mylocaluser mylocalgroup 1704 Jul  7 07:49 privkey1.pem
```

Where `ls -l` `/tmp/dms/customcert`:

```
-rw-r--r-- 1 998 1000 1846 Jul  7 07:49 cert1.pem
-rw------- 1 998 1000 1704 Jul  7 07:49 privkey1.pem
```




I changed the perms of `/etc/dms/tls/`:

```
-rw-r--r-- 1 docker docker 1846 Jul 10 19:40 cert
-rw------- 1 docker docker 1704 Jul 10 19:40 key
```


and tried again:

```
docker@mail:/etc/dovecot/conf.d$ doveadm auth test -a /var/spool/postfix/private/auth testuser testpass
```


to get that result:


```
Error: auth-client: conn unix:/var/spool/postfix/private/auth: connect(/var/spool/postfix/private/auth) failed: Permission denied (euid=5000(docker) egid=5000(docker) missing +x perm: /var/spool/postfix/private, dir owned by 101:0 mode=0700)
```

`ls -l` `/var/spool/postfix/`

```
drwx------ 2 postfix root     4096 Jul 10 19:40 private
```


confusion completed... 


regards

**Comment by smargold476** [comment]
Got it, in the end i found the solution:
`/etc/dovecot/userdb` --> username in DB  = `test@example.com` while as you can see in the screenshot thunderbird tries to login with user = `test` ...

Added `@example.com` and everything is working now .... Thanks to all for your help and I'll go and make some MR for updating the docs  :-)

**Comment by polarathene** [comment]
> Maybe the certificates are the problem after all?
> Or is that okay because of `/etc/dovecot/conf.d/10-ssl.conf`:

The permissions are fine.

The private key is meant to only be available to root user (_if someone has this, they can pretend to be servers for the domains in the certificate_), while the certificate is fine for anyone to read.

You're just running into permission issues as you're not running as the root user? (_Or the dovecot commands aren't_)

> I changed the perms of `/etc/dms/tls/`:

Don't do that.

---

> `Error: auth-client: conn unix:/var/spool/postfix/private/auth: connect(/var/spool/postfix/private/auth) failed: Permission denied (euid=5000(docker) egid=5000(docker) missing +x perm: /var/spool/postfix/private, dir owned by 101:0 mode=0700)`

I'm not sure if that's an actual problem, but I do recall someone having [similar issues with rootless Podman that I tried to investigate a bit](https://github.com/docker-mailserver/docker-mailserver/issues/2630#issuecomment-1151728893), but the user decided to switch to an alternative project instead of help resolve it.

---

> Got it, in the end i found the solution:
> `/etc/dovecot/userdb` --> username in DB = `test@example.com` while as you can see in the screenshot thunderbird tries to login with user = `test` ...

Ah, good find! Yeah I forgot about that one as I am rarely logging in manually. I remember getting confused by that the first time, as did many users, I don't think recall if our docs point that out somewhere.

> Thanks to all for your help and I'll go and make some MR for updating the docs :-)

You're welcome! Having some improvements to the docs would be hugely appreciated, thank you! :heart:
